### ISSUE TITLE
durableClient.CleanEntityStorageAsync() throws NRE, if durableClient was created with IDurableClientFactory.CreateClient()

### ISSUE THREAD

**Issue by scale-tone** [Azure/azure-functions-durable-extension#1958]
When using an IDurableClient instance injected via handler parameters in a normal way, everything works.

When obtaining that instance via IDurableClientFactory.CreateClient():
```
var durableClient = this._durableClientFactory.CreateClient(new DurableClientOptions
{
    TaskHub = taskHubName
});
```
, its .CleanEntityStorageAsync() method throws NullReferenceException with the following stack:
```
   at Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableClient.Microsoft.Azure.WebJobs.Extensions.DurableTask.IDurableEntityClient.CleanEntityStorageAsync(Boolean removeEmptyEntities, Boolean releaseOrphanedLocks, CancellationToken cancellationToken) in D:\\a\\r1\\a\\azure-functions-durable-extension\\src\\WebJobs.Extensions.DurableTask\\ContextImplementations\\DurableClient.cs:line 561\r\n   
at DurableFunctionsMonitor.DotNetBackend.CleanEntityStorage.<>c__DisplayClass2_0.<<DfmCleanEntityStorageFunction>b__0>d.MoveNext() in C:\\projects\\github\\DurableFunctionsMonitor\\durablefunctionsmonitor.dotnetbackend\\Functions\\CleanEntityStorage.cs:line 43\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   
at DurableFunctionsMonitor.DotNetBackend.HttpHandlerBase.<>c__DisplayClass2_0.<<HandleAuthAndErrors>b__0>d.MoveNext() in C:\\projects\\github\\DurableFunctionsMonitor\\durablefunctionsmonitor.dotnetbackend\\Common\\HttpHandlerBase.cs:line 36\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   
at DurableFunctionsMonitor.DotNetBackend.Globals.HandleErrors(HttpRequest req, ILogger log, Func`1 todo) in C:\\projects\\github\\DurableFunctionsMonitor\\durablefunctionsmonitor.dotnetbackend\\Common\\Globals.cs:line 64
```

All other methods seem to work fine.

IDurableClientFactory instance being used is the default one, injected via ctor.

Changing DurableClientOptions.IsExternalClient doesn't make any difference.

Running under normal func host.

### App Details

- **Durable Functions extension version (e.g. v1.8.3)**: 2.5.1
- **Azure Functions runtime version (1.0 or 2.0)**: 3.0
- **Programming language used**: C#

**Comment by sebastianburckhardt** [comment]
My first guess was that something is not fully initialized/started by the time `CreateClient` is called. However, just looking at the code I can't really see anything that would explain this.

I tried to repro this in a simple function app on the latest bits but no success, it seems to always work (see code below for reference).

What are the circumstances under which you see this behavior? Does this happen for empty task hubs, or only if there is content?

----------------

FYI, this is how I tested:

```csharp
 public class MyHttpTrigger
    {
        private readonly IDurableClientFactory _factory;

        public MyHttpTrigger(IDurableClientFactory factory)
        {
            this._factory = factory;
        }

        [FunctionName("MyHttpTrigger")]
        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req)
        {
            var client = this._factory.CreateClient(new Microsoft.Azure.WebJobs.Extensions.DurableTask.Options.DurableClientOptions()
            {
                ConnectionName = "AzureWebJobsStorage",
                TaskHub = "DotNetSmokeTestV2",
            });

            await client.CleanEntityStorageAsync(true, true, CancellationToken.None);

            return new OkObjectResult("Response from function with injected dependencies.");
        }
    }
```

**Comment by sebastianburckhardt** [comment]
Thanks for the details. I can see now where the exception originates in `Client.cs`. The code tries to access the `DurableTaskExtension` object via the field  `this.config` which is null. Apparently this field is not always populated for some clients. 

I can fix this pretty easily for `CleanEntityStorageAsync` by accessing the relevant information in another way. It does worry me a bit that this field seems to be accessed by other methods without null check, though.

**Comment by sebastianburckhardt** [comment]
fixed via #1971

**Comment by Ozzian** [comment]
When do we expect to see this released or is it in another package? The `Microsoft.Azure.WebJobs.Extensions.DurableTask` is still in 2.5.1
